waspc/data/Generator/templates/server/src/core/auth.js

```diff
@@ -69,5 +70,19 @@ export const verifyPassword = async (hashedPassword, password) => {
   }
 }
 
-export default auth
+// Looks up a user by email and if:
+// (1) the user does not exist, create them with a random password, or
+// (2) if the user does exist, update their existing password to a random password.
+export async function upsertUserWithRandomPassword(email) {
+  const randomPassword = uuidv4()
+
+  const user = await prisma.{= userEntityLower =}.upsert({
+    where: { email },
+    update: { password: randomPassword },
+    create: { email, password: randomPassword },
+  })
 
+  return user
```

Comment:
Yeah, I mainly left it as `user` explicitly for the readability. I find it easier to see variable names than try to parse the mustache template stuff. This will get nixed too after `UsernameAndPassword` at least. 👍🏻